### PR TITLE
[tool] In `flutter doctor -v`, warn when Android Studio version could not be detected.

### LINKED ISSUE #122081: `flutter doctor` should explicitly say when it can't find Android Studio version
When the tool isn't able to identify what version an Android Studio installation is, that can cause the build to fail because some logic depends on the Android Studio version. For example:
* #121468 
* #121925 

So when someone's running `flutter doctor -v` because things have gone wrong, the fact that the Android Studio version couldn't be identified is potentially a quite important clue for resolving the problem.

Currently, though, the `flutter doctor` output just suppresses mention of the version when it doesn't know the answer:
```dart
    // in [AndroidStudioValidator.validate]
    final String? studioVersionText = _studio.version == Version.unknown
      ? null
      : userMessages.androidStudioVersion(_studio.version.toString());
```

The result is a line like
```
[!] Android Studio
```

instead of like
```
[✓] Android Studio (version 2022.3)
```

This is too subtle; e.g. in #121925 it wasn't noticed in the first few rounds of debugging. It should instead say something like:
```
[!] Android Studio (unknown version)
```
or perhaps somewhat louder than that.

### PR DESCRIPTION
**Pull request by andrewkolos**
Fixes #122081.

When validating an Android Studio installation, add a warning validation message when we are unable to detect the version. This is because we have logic throughout the tool (JDK/JRE-searching) that is at higher risk of failing when we don't know the version.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by eliasyishak** [comment]
Is this ready for review? I'm noticing two tests that are failing but they could be flakes?

**Comment by andrewkolos** [comment]
> Is this ready for review? I'm noticing two tests that are failing but they could be flakes?

Yup. These were devicelab issues

File: packages/flutter_tools/lib/src/android/android_studio_validator.dart [hunk 1]
```diff
@@ -45,11 +50,11 @@ class AndroidStudioValidator extends DoctorValidator {
     final List<ValidationMessage> messages = <ValidationMessage>[];
     ValidationType type = ValidationType.missing;
 
-    final String? studioVersionText = _studio.version == null
-      ? null
-      : userMessages.androidStudioVersion(_studio.version.toString());
+    final String studioVersionText = _studio.version == null
+      ? 'version unknown'
+      : _userMessages.androidStudioVersion(_studio.version.toString());
     messages.add(ValidationMessage(
```
Comment (eliasyishak) [inline packages/flutter_tools/lib/src/android/android_studio_validator.dart:56]: Do we want to use the `_userMessages.androidStudioVersion()` with the string for version set to `unknown`? Seems like they both prefix `version` where needed

```suggestion
    final String studioVersionText = _studio.version == null
      ? _userMessages.androidStudioVersion('unknown')
      : _userMessages.androidStudioVersion(_studio.version.toString());
    messages.add(ValidationMessage(
```

File: packages/flutter_tools/lib/src/vscode/vscode_validator.dart [hunk 2]
```diff
@@ -23,13 +23,20 @@ class VsCodeValidator extends DoctorValidator {
...
   @override
   Future<ValidationResult> validate() async {
-    final String? vsCodeVersionText = _vsCode.version == null
-        ? null
+    final List<ValidationMessage> validationMessages =
+      List<ValidationMessage>.from(_vsCode.validationMessages);
+
+    final String vsCodeVersionText = _vsCode.version == null
+        ? 'unknown version'
         : userMessages.vsCodeVersion(_vsCode.version.toString());
```
Comment (eliasyishak) [inline packages/flutter_tools/lib/src/vscode/vscode_validator.dart:32]: Same thing here if the strings end up being the same thing

```suggestion
    final String vsCodeVersionText = _vsCode.version == null
        ? userMessages.vsCodeVersion('unknown')
        : userMessages.vsCodeVersion(_vsCode.version.toString());

```

Comment (eliasyishak) [reply packages/flutter_tools/lib/src/vscode/vscode_validator.dart:32]: Also, small nit but noticed that we are accessing `userMessages` differently between Android Studio (injected in constructor) and VS Code (referencing top level variable in `user_messages.dart`).

**Review by eliasyishak — APPROVED** [review]
lgtm!

**Pull request merged**